### DISCUSSION TITLE
Adopt the DDEX standard in the protocol

### DISCUSSION THREAD

**Discussion by web3rev** [originalworks/protocol#3] [General]
Documentation and examples of the ERN
[NFT-3226 - WD Music NFT 2024-04-30.zip](https://github.com/originalworks/protocol/files/15502391/NFT-3226.-.WD.Music.NFT.2024-04-30.zip)

**Comment by web3rev** [comment]
@0xbarath and @denciu . Please take a look at this documentation so we can discuss it's implementation.

**Reply by revelator-labs** [reply]
In order to comply with the DDEX naming convention I will map all of the fields we have defined as relevant to the Protocol data model and and adopt the DDEX filed names. 

In addition, we should probably allow our parsers to accept all DDEX fields so we can index them in the future but I would try to keep the requirements only to the bare minimum fields required to describe a musical work or recording and make sure they can be easily parsed by the protocol when validating the registration. 

The rest of the DDEX object should be accessible as additional data that can be posted by each data provider to their local IPFS node where additional proofs can be generated on-demand.

**Comment by criadoperez** [comment]
Sample implementation message both for ERN and MWN are now in the repository.
